lib/active_shipping/carriers/fedex.rb

```diff
@@ -153,8 +161,131 @@ def find_tracking_info(tracking_number, options = {})
       parse_tracking_response(xml, options)
     end
 
+
+    # Get Shipping labels
+    # Caveats:
+    #  - Only supports singlular packages
+    #
+    def create_shipment(origin, destination, packages, options = {})
+      options = @options.update(options)
+      packages = Array(packages)
+
+      begin
+        request = build_shipment_request(origin, destination, packages, options)
+        logger.debug(request) if logger
+
+        logger.debug(confirm_response) if logger
+
+        response = commit(save_request(request), (options[:test] || false))
+        parse_ship_response(response)
+      rescue RuntimeError => e
+        raise "Could not obtain shipping label. #{e.message}."
+      end
+    end
+
     protected
 
+    def build_shipment_request(origin, destination, packages, options = {})
+      imperial = %w(US LR MM).include?(origin.country_code(:alpha2))
```

Comment:
Maybe extract this logic into a helper instead of having it copy-pasted many times in the file.